azpubsub/src/main/scala/com/microsoft/azpubsub/security/auth/AzPubSubAclAuthorizer.scala

```diff
@@ -41,31 +38,47 @@ class AzPubSubAclAuthorizer extends SimpleAclAuthorizer with Logging with KafkaM
     explicitMetricName("azpubsub.security", "AuthorizerMetrics", name, metricTags)
   }
 
-  override def authorize(session: Session, operation: Operation, resource: Resource): Boolean = {
-    if (resource.resourceType == Topic && authZConfig.isDisabled(resource.name)) {
-      authorizerLogger.debug(s"AuthZ is disabled for resource: $resource")
-      successRate.mark()
-      disabledRate.mark()
-      return true
-    }
+  override def authorize(requestContext: AuthorizableRequestContext, actions: util.List[Action]): util.List[AuthorizationResult] = {
+    actions.asScala.map { action => {
+      val resource = AuthorizerUtils.convertToResource(action.resourcePattern)
+      if (resource.resourceType == Topic && authZConfig.isDisabled(resource.name)) {
+        aclAuthorizerLogger.debug(s"AuthZ is disabled for resource: $resource")
+        successRate.mark()
+        disabledRate.mark()
+        AuthorizationResult.ALLOWED
+      }
+
+      def getClaimRequestContext(requestContext: AuthorizableRequestContext, claimPrincipal: KafkaPrincipal): AuthorizableRequestContext = {
+        new AuthorizableRequestContext {
+          override def clientId(): String = requestContext.clientId
+          override def requestType(): Int = requestContext.requestType
+          override def listenerName(): String = requestContext.listenerName
+          override def clientAddress(): InetAddress = requestContext.clientAddress
+          override def principal(): KafkaPrincipal = claimPrincipal
+          override def securityProtocol(): SecurityProtocol = requestContext.securityProtocol
+          override def correlationId(): Int = requestContext.correlationId
+          override def requestVersion(): Int = requestContext.requestVersion
+        }
+      }
 
-    val sessionPrincipal = session.principal
-    if (classOf[AzPubSubPrincipal] == sessionPrincipal.getClass) {
-      val principal = sessionPrincipal.asInstanceOf[AzPubSubPrincipal]
-      for (role <- principal.getRoles.asScala) {
-        val claimPrincipal = new KafkaPrincipal(principal.getPrincipalType(), role)
-        val claimSession = new Session(claimPrincipal, session.clientAddress)
-        if (super.authorize(claimSession, operation, resource)) {
-          successRate.mark()
-          return true
+      val sessionPrincipal = requestContext.principal
+      if (classOf[AzPubSubPrincipal] == sessionPrincipal.getClass) {
+        val principal = sessionPrincipal.asInstanceOf[AzPubSubPrincipal]
+        for (role <- principal.getRoles.asScala) {
+          val claimPrincipal = new KafkaPrincipal(principal.getPrincipalType(), role)
+          val claimRequestContext = getClaimRequestContext(requestContext, claimPrincipal)
+          if (super.authorize(claimRequestContext, List(action).asJava).asScala.head == AuthorizationResult.ALLOWED) {
+            successRate.mark()
+            AuthorizationResult.ALLOWED
```

Comment:
better to add return statement in all place.

In Scala, if there is no return then the last expression is taken to be the return value. So, if the last expression is what you want to return, then you can omit the return keyword. But if what you want to return is not the last expression, then Scala will not know that you wanted to return it.